pkg/image/fetcher.go

```diff
@@ -109,7 +110,17 @@ func (f *Fetcher) Fetch(ctx context.Context, name string, options FetchOptions)
 	}
 
 	f.logger.Debugf("Pulling image %s", style.Symbol(name))
-	err = f.pullImage(ctx, name, options.Platform)
+	platform := options.Platform
+	for i := 0; i <= 1; i++ {
+		err = f.pullImage(ctx, name, platform)
+		if err != nil && errors.Is(err, ErrPlatformNotMatch) {
+			f.logger.Warnf("%s, retrying without platform", err.Error())
```

Comment:
But since version `0.32` `pack build` seems to be broken always showing `does not match the specified platform: wanted linux/amd64, actual: linux`

Wouldn't that mean that there will always be a warning? For every buildpack. Is that the desired UX?

Reply:
> Is that the desired UX?

Probably not. We could make it an info message, or just omit it.

Reply:
> With that, there could be an info before 2. and a warning before 3. WDYT?

This makes sense to me

Reply:
I added the **warning** because we are working to improve the buildpack package creation and set the `os/arch`.
I was expecting in short term, we will start having buildpacks with the `os/arch` defined and the warning will be not be shown.

Reply:
For now, I just removed the **warning** message, because adding `Try (os)` will imply to parse the given platform, we still have work to do on this, but I think falling back to the default behavior is good to fix the bug.